lib/distribute.h

```diff
@@ -38,21 +38,39 @@ struct distribute {
 	/* Name of the interface. */
 	char *ifname;
 
+	/* Name of the interface. */
+	char *vrfname;
+
 	/* Filter name of `in' and `out' */
 	char *list[DISTRIBUTE_MAX];
 
 	/* prefix-list name of `in' and `out' */
 	char *prefix[DISTRIBUTE_MAX];
 };
 
+struct distribute_ctx {
+	/* Hash of distribute list. */
+	struct hash *disthash;
+
+	/* Hook functions. */
+	const char *(*distribute_get_vrf_fn)(void *ctx);
+	void (*distribute_add_hook)(void *ctx, struct distribute *dist);
+	void (*distribute_delete_hook)(void *ctx, struct distribute *dist);
```

Comment:
these hooks can probably be global variables like they were before, because all routing instances from a single daemon should use the same hooks.

Reply:
if you don't mind, I wil keep it in the structure.